### PR TITLE
Add basic unit tests for command line parsing

### PR DESCRIPTION
**Pull request by IRCody**
*Description of changes:*
Adds some unit tests for the parsing functions. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+		return false
+	}
+	return true
+}
+
+func equalString(a, b *string) bool {
+	if a == nil && b == nil {
+		return true
+	}
+	if (a == nil && b != nil) || (a != nil && b == nil) {
```
Comment (mxpv) [inline main_test.go]: having `if a == nil && b == nil` we can write `else if a == nil || b == nil {`

File: main_test.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+			in:        []string{tempFile.Name() + ":rw"},
+			outDrives: []models.Drive{validDrive},
+			expectedErr: func(a error) bool {
+				return a == nil
+			},
+		},
+	}
+	for _, c := range cases {
+		drives, err := parseBlockDevices(c.in)
+		if !equalDrives(c.outDrives, drives) {
```
Comment (xibz) [inline main_test.go]: would probably be better to just use `reflect.DeepEqual` here.

File: main_test.go [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+			in:         []string{"a:b"},
+			outDevices: []firecracker.VsockDevice{},
+			expectedErr: func(a error) bool {
+				return a == unableToParseVsockCID
+			},
+		},
+	}
+	for _, c := range cases {
+		devices, err := parseVsocks(c.in)
+		if !equalVsockDevices(devices, c.outDevices) {
```
Comment (xibz) [inline main_test.go]: same here.

File: main_test.go (hunk 1 above)
Comment (IRCody) [reply main_test.go]: Code was removed in refactor.

File: main_test.go (hunk 2 above)
Comment (IRCody) [reply main_test.go]: Yea that is definitely a lot better, thanks!

**Pull request merged**